### ISSUE TITLE
#if stache helper double runs computes even with @

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-stache#186]
@dylanrtt commented on [Thu Sep 08 2016](https://github.com/canjs/canjs/issues/2480)

Using the `#if` stache helper will run a function argument even if prefixed with `@`.

```
{{#if @foo}} true {{/if}}
```

http://jsbin.com/yifagodeco/edit?js,console,output

The cause seems to be that `can.compute.truthy()` will run the compute's result if it is a function.

https://github.com/canjs/canjs/blob/b26e84eb141c5cfa9914d8c8f6a578b053ce6207/compute/compute.js#L69

```
can.compute.truthy = function (compute) {
    return can.compute(function () {
        var res = compute();
        if (typeof res === 'function') {
            res = res();
        }
        return !!res;
    });
};
```

**Comment by christopherjbaker** [comment]
This looks to be a duplicate of #172, which was fixed by #178.

**Comment by justinbmeyer** [comment]
yeah, duplicate, closing for #172
